doc/nrf/ug_nrf91.rst

```diff
@@ -3,11 +3,411 @@
+      In its default configuration, the DFU target library is set to support all the types of FOTA upgrades except full modem firmware upgrades, but you can freely enable or disable the support for specific targets.
+
+      In addition, the following requirements apply:
+
+      * |fota_upgrades_req_mcuboot|
+      * If you want to upgrade the upgradable bootloader, the :ref:`bootloader` must be used (:kconfig:`CONFIG_SECURE_BOOT`).
+      * If you want to upgrade the modem firmware through modem delta updates, neither MCUboot nor the immutable bootloader are required, because the modem firmware upgrade is handled by the modem itself.
+      * If you want to perform a full modem firmware upgrade, an |external_flash_size| is required.
+
+#. Create a binary file that contains the new image.
+
+      .. note::
+         This step does not apply for upgrades of the modem firmware.
+         You can download delta patches and full binaries of the modem firmware from the `nRF9160 product website (compatible downloads)`_.
+
+      |fota_upgrades_building|
+      The :file:`app_update.bin` file is the file that should be uploaded to the server.
+
+      To create binary files for a bootloader upgrade, make sure that :kconfig:`CONFIG_SECURE_BOOT` and :kconfig:`CONFIG_BUILD_S1_VARIANT` are enabled and build MCUboot as usual.
+      The build will create a binary file for each variant of the upgradable bootloader, one for each bootloader slot.
+      See :ref:`upgradable_bootloader` for more information.
+
+#. Make the binary file (or files) available for download.
+     Upload the serialized :file:`.cbor` binary file or files to a web server that is compatible with the :ref:`lib_download_client` library.
+     One way of doing this is to upload the files to an Amazon Web Services Simple Storage Service (AWS S3) bucket.
+     See the :ref:`lib_aws_fota` documentation for instructions.
+
+     Your application must be able to retrieve the host and file name for the binary file.
+     See :ref:`lib_fota_download` for information about the format of this information, especially when providing two files for a bootloader upgrade.
+     You can hardcode the information in the application, or you can use functionality like AWS jobs to provide the URL dynamically.
+
+The full FOTA procedure depends on where the binary files are hosted for download.
+
+You can refer to the following implementation samples:
+
+* :ref:`http_full_modem_update_sample` - performs a full firmware OTA update of the modem.
+* :ref:`http_modem_delta_update_sample` - performs a delta OTA update of the modem firmware.
+* :ref:`http_application_update_sample` - performs a basic application FOTA update.
+* :ref:`aws_fota_sample` - performs a FOTA update via MQTT and HTTP, where the firmware download is triggered through an AWS IoT job.
```

Comment:
Asset Tracker v1 and v2 also support FOTA updates of APP, MODEM (delta), and BOOT (if enabled in .cfg).